File: MartianRobots.Data/Entities/RobotStep.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;
using MartianRobots.Models;
using MartianRobots.Models.Constants;

namespace MartianRobots.Data.Entities
{
    public class RobotStep
    {
        [BsonId]
        public ObjectId Id { get; set; }
        [BsonRepresentation(BsonType.String)]
        public Guid RunId { get; set; }
        public int RobotId { get; set; }
        public int StepNumber { get; set; }
        public Position Position { get; set; }
        public OrientationState Orientation { get; set; }
        public RectangularMoveCommand? Command { get; set; }
        public bool IsLost { get; set; }
        public bool IsLastStep { get; set; }
    }
}

File: ui/src/components/Model/RobotStep.ts
import { Position, OrientationState } from "./Position";

export interface RobotStep {
    robotId: number,
    stepNumber: number,
    position: Position,
    orientation: OrientationState,
    command?: RectangularMoveCommand,
    isLost: boolean,
    isLastStep: boolean
}

export enum RectangularMoveCommand {
    Left = 1,
    Right = 2,
    Forward = 3
}

export function getRectangularMoveCommandName(command: RectangularMoveCommand): string {
    switch (command)
    {        
        case RectangularMoveCommand.Left:
            return "Left";
        case RectangularMoveCommand.Right:
            return "Right";
        case RectangularMoveCommand.Forward:
            return "Forward";
        default:
            return "";
    }
}